src/app/containers/ConsentBanner/Banner/index.canonical.jsx

```diff
@@ -45,6 +45,14 @@ const CanonicalConsentBannerContainer = ({ type, onReject, onAccept }) => {
 
   const dataAttribute = getDataAttribute(type);
 
+  const consentBannerRef = useRef(null);
+
+  useEffect(() => {
+    if (consentBannerRef) {
+      consentBannerRef.current.focus();
+    }
+  });
```

Comment:
This hook will run not only on mount and unmount, but also on every render. If something causes this component to be re-rendered, focus will be stolen from the user and set back to the beginning. Some of the pain of this could be managed by using an empty dependencies array. E.G.

```jsx
useEffect(() => {
    if (consentBannerRef) {
      consentBannerRef.current.focus();
    }
  }, []);
```

Reply:
good point! Would adding a `const [ focused, setFocused] = useState(false);` be more readable? How does the empty dependency array work?

Reply:
Just read up on it, I think `useEffect` is executed once on mount and then whenever a dependency changes value, so with no dependencies it only runs on mount, sounds good